src/m365/aad/commands/group/group-list.ts

```diff
@@ -0,0 +1,51 @@
+import { Logger } from '../../../../cli';
+import GlobalOptions from '../../../../GlobalOptions';
+import { GraphItemsListCommand } from '../../../base/GraphItemsListCommand';
+import commands from '../../commands';
+import { Group } from '@microsoft/microsoft-graph-types';
+
+interface CommandArgs {
+  options: GlobalOptions;
+}
+
+class AadGroupListCommand extends GraphItemsListCommand<Group>   {
+  public get name(): string {
+    return commands.GROUP_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all groups defined in Azure Active Directory.';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'displayName', 'groupType'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getAllItems(`${this.resource}/v1.0/groups`, logger, true)
+      .then((): void => {
+        if (args.options.output === 'text') {
+          this.items.forEach((group: any) => {
```

Comment:
Let's avoid using `any` to avoid runtime and refactoring issues in the future. Since we know the exact shape of the data that we get, let's use the specific type instead.